curriculumBuilder/settings.py

```diff
@@ -421,26 +423,55 @@
-                         'textarea', 'tfoot', 'th', 'thead', 'tr', 'tt', '', 'ul', 'var', 'wbr', 'summary', 'details')
-RICHTEXT_ALLOWED_STYLES = ('margin-top', 'margin-bottom', 'margin-left', 'margin-right', 'float', 'vertical-align',
-                           'border', 'margin', 'width', 'height', 'max-width', 'padding', 'margin', 'style',
-                           'data-pdf-link', 'data-lightbox', 'data-title', 'color', 'background', 'background-color')
-RICHTEXT_ALLOWED_ATTRIBUTES = (
-    'abbr', 'accept', 'accept-charset', 'accesskey', 'action', 'align', 'alt', 'axis', 'border',
-    'cellpadding', 'cellspacing', 'char', 'charoff', 'charset', 'checked', 'cite', 'class',
-    'clear', 'cols', 'colspan', 'color', 'compact', 'coords', 'data', 'datetime', 'dir', 'disabled',
-    'enctype', 'for', 'frame', 'headers', 'height', 'href', 'hreflang', 'hspace', 'id',
-    'ismap', 'label', 'lang', 'longdesc', 'maxlength', 'media', 'method', 'multiple', 'name',
-    'nohref', 'noshade', 'nowrap', 'open', 'prompt', 'readonly', 'rel', 'rev', 'role','rows', 'rowspan',
-    'rules', 'scope', 'selected', 'shape', 'size', 'span', 'src', 'start', 'style', 'summary',
-    'tabindex', 'target', 'title', 'type', 'usemap', 'valign', 'value', 'vspace', 'width',
-    'xml:lang', 'data-pdf-link', 'data-lightbox', 'data-title', 'data-start', 'data-end', 'aria-expanded')
+RICHTEXT_ALLOWED_TAGS = (
+    'a', 'abbr', 'acronym', 'address', 'area', 'article', 'aside', 'b', 'bdo', 'big', 'blockquote',
+    'br', 'button', 'caption', 'center', 'cite', 'code', 'col', 'colgroup', 'dd', 'del', 'dfn',
+    'dir', 'div', 'display', 'dl', 'dt', 'em', 'fieldset', 'figure', 'font', 'footer', 'form', 'h1',
+    'h2', 'h3', 'h4', 'h5', 'h6', 'header', 'hr', 'i', 'iframe', 'img', 'input', 'ins', 'kbd',
+    'label', 'legend', 'li', 'map', 'men', 'nav', 'object', 'ol', 'optgroup', 'option', 'p', 'pre',
+    'q', 's', 'samp', 'section', 'select', 'small', 'span', 'strike', 'strong', 'sub', 'sup',
+    'table', 'tbody', 'td', 'textarea', 'tfoot', 'th', 'thead', 'tr', 'tt', '', 'ul', 'var', 'wbr',
+    'summary', 'details'
+)
+RICHTEXT_ALLOWED_STYLES = (
+    'margin-top', 'margin-bottom', 'margin-left', 'margin-right', 'float', 'vertical-align',
+    'border', 'margin', 'width', 'height', 'max-width', 'padding', 'margin', 'style',
+    'data-pdf-link', 'data-lightbox', 'data-title', 'color', 'background', 'background-color'
+)
+
+def iframe_filter_domain(name, value):
+    """
+    We use iframes in curriculum descriptions to embed Google Docs containing
+    the lesson plans. We want to continue to allow iframes in general, but
+    restrict to only allow google docs.
+
+    If in the future we want to expand our usage of iframes to include other use
+    cases, those should be simililarly captured here.
+    """
+    if name in ('width', 'height', 'frameborder', 'style'):
```

Comment:
For my own education, why is this necessary?

Reply:
this function will override the '*' allowed tags, so we have to make sure that the attributes we want to allow are explicitly allowed here